langlib/lang.value/src/main/ballerina/value.bal

```diff
@@ -153,6 +186,10 @@ public isolated function toBalString(any v) returns string = @java:Method {
 # The subset of Ballerina expression syntax supported is that produced
 # by toBalString when applied to an anydata value.
 #
+# ```ballerina
+# anydata year = check "2023".fromBalString();
```

Comment:
Shall we add something like
`int year = <int> check "2023".fromBalString();`
It may be bit complicate, but in common scenarios users need to convert this to int, not to anydata.

Reply:
Let's not use casts. If the user wanted to get the value as an `int`, they should ideally use `ensureType` or an `is` check.

Reply:
Shall we change this example to use like an array, map, etc.?

To parse a string as an int, we would generally use `int:fromString`.